go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -271,29 +274,68 @@ func (c *Controller) syncServices(key string) error {
 
 	vips := collectServiceVIPs(service)
 	if svcNeedsIdling(service.Annotations) && !util.HasValidEndpoint(service, endpointSlices) {
+		// addServiceToIdlingBalancer adds the vips to service tracker
 		err = c.addServiceToIdlingBalancer(vips, service)
 		if err != nil {
 			c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToAddToIdlingBalancer",
 				"Error trying to add to Idling LoadBalancer for Service %s/%s: %v", name, namespace, err)
 			return err
 		}
-		vipsTracked = vipsTracked.Delete(vips.List()...)
-		err = deleteVIPsFromNonIdlingOVNBalancers(vips, name, namespace)
+		// create a temporary map to convert sets.String to map for passing to functions
+		tmpVIPs := make(map[string]string)
+		for v := range vips {
+			delete(vipsTracked, v)
+			tmpVIPs[v] = ""
+		}
+		needsCleanUp := false
+		for vipProtocol := range vips {
+			vip, _ := splitVirtualIPKey(vipProtocol)
+			if c.serviceTracker.getLoadBalancer(name, namespace, vip) != loadbalancer.IdlingLoadBalancer {
+				needsCleanUp = true
+				break
+			}
+		}
+		if needsCleanUp {
+			err = deleteVIPsFromNonIdlingOVNBalancers(tmpVIPs, name, namespace)
+			if err != nil {
+				c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToDeleteOVNLoadBalancer",
+					"Error trying to delete the OVN LoadBalancer while setting up Idling for Service %s/%s: %v", name, namespace, err)
+				return err
+			}
+		}
+		// at this point we have processed all vips we've found in the service
+		// so the remaining ones that we had in the vipsTracked variable should be deleted
+		// We remove them from OVN and from the tracker
+		err = deleteVIPsFromAllOVNBalancers(vipsTracked, name, namespace)
```

Comment:
did we processed all vips here?
I think this was an `		err = deleteVIPsFromNonIdlingOVNBalancers(vips, name, namespace)` before

Reply:
the delete from non idling still happens above, but if we get here we are going to return out on line 317, and we were missing cleaning up OVN, but we were deleting the vips from the tracker. I broke this out into its own commit to make this less confusing (and ensure we dont revert this fix in the future)